client/components/domains/register-domain-step/index.jsx

```diff
@@ -263,9 +263,23 @@ class RegisterDomainStep extends Component {
 		this._isMounted = false;
 	}
 
+	// In the launch flow, the initial query could sometimes be missing if the user had
+	// created a site by skipping the domain step. In these cases, fire the initial search
+	// with the subdomain name.
+	getInitialQueryInLaunchFlow() {
+		if ( ! this.props.isInLaunchFlow ) {
+			return;
+		}
+
+		const wpcomSubdomainWithRandomNumberSuffix = /^(.+?)([0-9]{5,})\.wordpress\.com$/i;
+		const [ , strippedHostname ] =
+			this.props.selectedSite.domain.match( wpcomSubdomainWithRandomNumberSuffix ) || [];
```

Comment:
It looks like `this.props.selectedSite` may be a boolean according to the prop type declaration, so it may be worth adding a defensive early exit condition to make sure we have an object and that the `domain` property is a string.
```suggestion
		if ( typeof this.props.selectedSite !== 'object' || typeof this.props.selectedSite.domain !== 'string' ) {
			return;
		}

		const wpcomSubdomainWithRandomNumberSuffix = /^(.+?)([0-9]{5,})\.wordpress\.com$/i;
		const [ , strippedHostname ] =
			this.props.selectedSite.domain.match( wpcomSubdomainWithRandomNumberSuffix ) || [];
```